### ISSUE TITLE
Invalid emit for destructuring with private identifier

### ISSUE THREAD

**Issue by ajafff** [microsoft/TypeScript#35936]
**TypeScript Version:**  3.8.0-dev.20191228

**Search Terms:**

**Code**

```ts
class C{
    #prop = 1;
    method() {
        const { #prop: prop } = this;
        ({ #prop: this.#prop } = this);
    }
}
```

**Expected behavior:**

No syntax error in emitted code.

**Actual behavior:**

```js
    method() {
        var _a;
        const { : prop } = this;
        (_a = this, { : ({ set value(_b) { __classPrivateFieldSet(_a, _prop, _b); } }).value } = this);
    }
```

**Playground Link:** https://www.typescriptlang.org/play/?ts=3.8.0-dev.20191228&ssl=7&ssc=2&pln=1&pc=1#code/MYGwhgzhAEDCDeAoaLoGIAOAnA9h6AvNAIwDcyqAtgKYAuAFjgCYAUAlNEqt9MDgHYRandNjwAuaGPwBfQtAYBLCOR6oW8UbgySlEAHSZt0OUT1tVqGYhlA

**Related Issues:** #35934 #35935  /cc @mheiber

**Comment by dragomirtitian** [comment]
Are you sure the error here isn't that a syntax error is not issued in TS on those private names? Both babel ([ex](https://babeljs.io/repl#?browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYGwhgzhAEDCDeAoaLoGIAOAnA9h6AvNAIwDcyqAtgKYAuAFjgCYAUAlNEqt9MDgHYRandNjwAuaGPwBfQtAYBLCOR6oW8UbgySlEAHSZt0OUT1tVqGYmuJ-1AO5x2-mg2bsgA&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=true&timeTravel=false&sourceType=module&lineWrap=true&presets=es2015%2Creact%2Cstage-1%2Cstage-2%2Cstage-3&prettier=true&targets=&version=7.7.7&externalPlugins=)) and node 13.6 issue syntax errors on those.

**Comment by ajafff** [comment]
Wow, it didn't come to my mind that destructuring private names would not be allowed.

Relevant discussion in the proposal repo: https://github.com/tc39/proposal-class-fields/issues/4

TL;DR: since structuring object literals with private named properties is not allowed, **de**structuring is also not possible.
